### DISCUSSION TITLE
关于SQL指令的具体实现

### DISCUSSION THREAD

**Discussion by Randy-13** [liusy58/NJU_AdvancedProgramming_SP22#85] [Q&A] [answered]
这里想再确认一下项目对于SQL指令的要求：
是例如：当用户选择“1.发布商品”时，
fstream  fs1
fs1.open(...ios::out);
if(select == 1)
{
    fs1 << "INSERT INTO commodity VALUES"<<endl;
}
fs1.close();
..........
fstream fs2
fs2.open(..ios::in)
string temp;
fs2>>temp;
if(temp ==  "INSERT INTO commodity VALUES")
{
   Seller_insert();//调用发布商品函数
}
是这样的意思嘛

**Comment by binbin00** [comment, marked as answer]
存到文件里只是为了方便查阅。你可以直接解析你合成的指令，但不能直接就 if(temp == "INSERT INTO commodity VALUES")吧，你怎么知道这个指令是INSERT INTO而不是别的？你怎么知道要操作的是哪个表？

**Comment by Randy-13** [comment]
谢谢学长的回答，但还是有几个地方不太理解，方便查阅：指的是把使用者在程序中的一些操作记录下来，相当于浏览痕迹?然后便于查阅?，解析我合成的指令：我写的if的意思是，我想用多个if，else，分类嘛，如果不用，那么学长的意思是如何解析呢?还有最后一个问题，你提到的“表”指的是什么呢( ᵒ̴̶̷̤໐ᵒ̴̶̷̤ )，问题可能有点多，，抱歉啦！

**Reply by binbin00** [reply]
1. 方便助教检查时查阅，原因是这块并不要求提供解析指令的接口（计算器模块是要求的，且要测试接口）
2. 至于解析指令的要求，最起码可以直接执行已有的正确的指令序列，进一步，你可以判断指令是否符合格式，更进一步...
3. ppt37页提到过。
有问题可以尽情提问，我们助教有责任且乐于回答。
